src/panoramic/cli/field/client.py

```diff
@@ -0,0 +1,121 @@
+        self.calculation = calculation
+        self.aggregation = aggregation
+
+    @classmethod
+    def from_dict(cls, inputs: Dict[str, Any]) -> 'Field':
+        return cls(
+            slug=inputs['slug'],
+            group=inputs['group'],
+            display_name=inputs['display_name'],
+            data_type=inputs['data_type'],
+            field_type=inputs['field_type'],
+            description=inputs.get('description'),
+            data_source=inputs.get('data_source'),
+            aggregation=inputs.get('aggregation'),
+            calculation=inputs.get('calculation'),
+        )
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {
+            'slug': self.slug,
+            'group': self.group,
+            'display_name': self.display_name,
+            'data_type': self.data_type,
+            'field_type': self.field_type,
+            'description': self.description,
+            'calculation': self.calculation,
+            'aggregation': self.aggregation,
+            'data_source': self.data_source,
+        }
+
+    def __hash__(self) -> int:
+        return hash(self.to_dict())
+
+    def __eq__(self, o: object) -> bool:
+        if not isinstance(o, Field):
+            return False
+
+        return self.to_dict() == o.to_dict()
+
+
+class FieldClient(OAuth2Client, VersionedClient):
+
+    """Model field HTTP API client."""
+
+    base_url: str
+
+    def __init__(
+        self, base_url: Optional[str] = None, client_id: Optional[str] = None, client_secret: Optional[str] = None,
+    ):
+        base_url = base_url if base_url is not None else get_base_url()
+        client_id = client_id if client_id is not None else get_client_id()
+        client_secret = client_secret if client_secret is not None else get_client_secret()
+
+        self.base_url = base_url
+        super().__init__(client_id, client_secret)
+
+    def create_fields(self, company_slug: str, fields: List[Field], data_source: Optional[str] = None):
+        params = {'company_slug': company_slug, 'virtual_data_source': data_source}
+        response = self.session.post(
+            self.base_url, params=params, json=[field.to_dict() for field in fields], timeout=30
+        )
+        response.raise_for_status()
+
+    def get_fields(
+        self, company_slug: str, data_source: Optional[str] = None, offset: int = 0, limit: int = 100
+    ) -> List[Field]:
+        """Retrieve all fields in a given source and optionally a datasource."""
+        logger.debug(f'Listing fields for company: {company_slug} source: {data_source}')
+        params = {'company_slug': company_slug, 'virtual_data_source': data_source, 'offset': offset, 'limit': limit}
+        response = self.session.get(self.base_url, params=params)
+        response.raise_for_status()
+        response_json = response.json()['data']
+        return [Field.from_dict(d) for d in response_json]
+
+    def update_fields(self, company_slug: str, fields: List[Field]):
```

Comment:
Like I mentioned elsewhere, we can get rid of this method and use `create_fields` only and rename it to `upsert_fields`

Reply:
Yeah as agreed - let's change it back.

Reply:
Fixed in 547db1e.